pkg/operators/localmanager/localmanager.go

```diff
@@ -0,0 +1,326 @@
+	containerSelector := containercollection.ContainerSelector{
+		Name: l.params.Get(ContainerName).AsString(),
+	}
+
+	if setter, ok := l.gadgetInstance.(MountNsMapSetter); ok {
+		// Create mount namespace map to filter by containers
+		mountnsmap, err := l.manager.localGadgetManager.CreateMountNsMap(containerSelector)
+		if err != nil {
+			return commonutils.WrapInErrManagerCreateMountNsMap(err)
+		}
+
+		log.Debugf("set mountnsmap for gadget")
+		setter.SetMountNsMap(mountnsmap)
+
+		l.mountnsmap = mountnsmap
+	}
+
+	if attacher, ok := l.gadgetInstance.(Attacher); ok {
+		l.attacher = attacher
+
+		attachContainerFunc := func(container *containercollection.Container) {
+			log.Debugf("calling gadget.AttachContainer()")
+			err := attacher.AttachContainer(container)
+			if err != nil {
+				log.Warnf("start tracing container %q: %s", container.Name, err)
+				return
+			}
+
+			l.attachedContainers[container] = struct{}{}
+
+			log.Debugf("tracer attached: container %q pid %d mntns %d netns %d",
+				container.Name, container.Pid, container.Mntns, container.Netns)
+		}
+
+		detachContainerFunc := func(container *containercollection.Container) {
+			log.Debugf("calling gadget.DetachContainer()")
+			err := attacher.DetachContainer(container)
+			if err != nil {
+				log.Warnf("stop tracing container %q: %s", container.Name, err)
+				return
+			}
+			log.Debugf("tracer detached: container %q pid %d mntns %d netns %d",
+				container.Name, container.Pid, container.Mntns, container.Netns)
+		}
+
+		id := uuid.New()
+		l.subscriptionKey = id.String()
+
+		log.Debugf("add subscription")
+		containers := l.manager.localGadgetManager.Subscribe(
+			l.subscriptionKey,
+			containerSelector,
+			func(event containercollection.PubSubEvent) {
+				log.Debugf("%s: %s", event.Type.String(), event.Container.ID)
+				switch event.Type {
+				case containercollection.EventTypeAddContainer:
+					attachContainerFunc(event.Container)
+				case containercollection.EventTypeRemoveContainer:
+					detachContainerFunc(event.Container)
+				}
+			},
+		)
+
+		for _, container := range containers {
+			attachContainerFunc(container)
+		}
+	}
+
+	return nil
+}
+
+func (l *localManagerTrace) PostGadgetRun() error {
+	if l.mountnsmap != nil {
+		log.Debugf("calling RemoveMountNsMap()")
+		l.manager.localGadgetManager.RemoveMountNsMap()
+	}
+	if l.subscriptionKey != "" {
+		log.Debugf("calling Unsubscribe()")
+		l.manager.localGadgetManager.Unsubscribe(l.subscriptionKey)
+
+		// emit detach for all remaining containers
+		for container := range l.attachedContainers {
+			l.attacher.DetachContainer(container)
+		}
+	}
+	return nil
+}
+
+func (l *localManagerTrace) enrich(ev any) error {
+	if _, canEnrichEventFromMountNs := ev.(operators.ContainerInfoFromMountNSID); canEnrichEventFromMountNs {
+		err := l.manager.localGadgetManager.ContainerCollection.EnrichEventByMntNs(ev)
+		if err != nil {
+			return err
+		}
```

Comment:
Well, it's not actually happening with that gadget because of the `canEnrichEventFromMountNs` check before. I'm wondering if those `EnrichEventBy*` functions should return an error.

Reply:
> Should it really return here?
>
Mmm good point. Given that it is a best-effort function where we try to complement the enrichers among them, it should simply go to the next one. Maybe it could be a `log.Warnf`?

> I'm wondering if those EnrichEventBy* functions should return an error.
>
I think they should in order to protect someone using the `container-collection` package directly. Even tough in the case of the `localmanager`, it is not helpful at the moment.

Reply:
>> I'm wondering if those EnrichEventBy* functions should return an error.

> I think they should in order to protect someone using the container-collection package directly. Even tough in the case of the localmanager, it is not helpful at the moment.

I removed the error from the `EnrichEventBy*` and replaced the parameters with the respective interfaces (so the errors can't occur in the first place, leaving casting up to the caller). That also prevents typecasting twice for both enrichers.